### ISSUE TITLE
Stack makes infinite height if only have Positioned children

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter#28610]
I run into a problem where I had a `Stack` with only `Positioned` children and my app was saying the `Stack` was generating an infinite size. I found the solution [here](https://stackoverflow.com/a/54624340/1227166), which is a little hack, to just place an empty `Container` inside the `Stack`, so it will know how to calculate its own size.

I believe the `Stack` documentation should be clearer about this use case and what to do to prevent it. Also `Stack` should have a way (maybe a flag?) to measure itself in case there are only `Positioned` children so we won't have to resort to these kind of hacks.

**Comment by Kartik1607** [comment]
Could you share the code which is causing the issue.

**Comment by feinstein** [comment]
This was inside a method, so there are some parameters referenced in the code.

```
Stack(
  alignment: Alignment.center,
  overflow: Overflow.visible,
  children: <Widget>[    
    // we need this to make the rounded logos match the rounded corners of the button
    Positioned(
  	  left: left ?? -8,
  	  child: Row(
  	    children: <Widget>[
  	  	  icon ??
  	  	  	SvgPicture.asset(
  	  	  	  iconAssetName,
  	  	  	  height: 24,
  	  	  	),
  	  	  SizedBox(width: 24),
  	  	    Text(
  	  	    	text,
  	  	    	style: textStyle
  	  	    ),
  	      ],
  	  ),
    ),
  ],
),
```

**Comment by MasiaVittorio** [comment]
Hi @feinstein, I'm trying to achieve something similar to what you're talking about I guess, but putting the container inside the stack just makes the infinite height error disappear (while the content of the stack istelf isn't visible since the whole stack just has zero height for some reason now).

How can you force the stack to just occupy the minimum amount of height needed to fit its only child?

**Comment by feinstein** [comment]
Hi @MasiaVittorio, this somewhat relates to my other issue #46803, I am busy right now, but see if  `InstrinsicHeight` helps you.

**Comment by PiN73** [comment]
Another way is to replace `Positioned` with `Padding`

**Comment by YatsenkoDev** [comment]
Wrap your Stack with Column mainAxisSize: MainAxisSize.min and it will magically fit his child.

**Comment by iapicca** [comment]
Hi @feinstein 
I'm not sure this behaviour is intentional, but it should at least be referenced in the docs
Thank you

**Comment by feinstein** [comment]
Hi @iapicca, I think this should be confirmed with @HansMuller if it's intentional behavior, otherwise we will be documenting bugs and promoting them to "intentional behavior".

**Comment by iapicca** [comment]
@feinstein I agree with you, I'm the first to state my lack of confidence on the matter

**Comment by PaulLVoelker** [comment]
@YatsenkoDev could you elaborate a bit? I couldn't follow you.

**Comment by rockingdice** [comment]
@polarby 
```
Column(
  mainAxisSize: MainAxisSize.min,
  children: [
    Stack(
      children: [Placeholder()]
    )
  ]
)
```

**Comment by nate-thegrate** [comment]
I'm interested in working on this issue, but I've been having trouble reproducing the bug.

I did my best to recreate the Stack that @feinstein shared, but the Stack widget no longer has an `overflow` parameter, so I used `clipBehavior: .none` instead.

Here's a [**DartPad sample**](https://dartpad.dev/?id=65c1f6b5ebf17432f1aea8fdce0853c0) that displays a Stack with a Positioned child. What should I change in order to make an error pop up?
